torchvision/csrc/cpu/video/Video.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#ifndef VIDEO_H_
+#define VIDEO_H_
+
+#include <map>
+#include <regex>
+#include <string>
+#include <vector>
+
+#include <ATen/ATen.h>
+#include <Python.h>
+#include <c10/util/Logging.h>
+#include <torch/script.h>
+
+#include <exception>
+#include "defs.h"
+#include "memory_buffer.h"
+#include "sync_decoder.h"
+
+using namespace ffmpeg;
+
+struct Video : torch::CustomClassHolder {
+  std::tuple<std::string, long> current_stream; // stream type, id
+  // global video metadata
+  c10::Dict<std::string, c10::Dict<std::string, std::vector<double, std::allocator<double>>>> streamsMetadata;
+
+ public:
+  Video(std::string videoPath, std::string stream, bool isReadFile);
+  std::tuple<std::string, int64_t> getCurrentStream() const;
+  c10::Dict<std::string, c10::Dict<std::string, std::vector<double, std::allocator<double>>>> getStreamMetadata() const;
+  void Seek(double ts);
+  bool setCurrentStream(std::string stream = "video");
```

Comment:
Can we avoid having a default value in the `h` file? I think it's better to be explicit in the call sites.

Reply:
Sure.